includes/internal/emails/class-email-list-table-actions.php

```diff
@@ -0,0 +1,98 @@
+<?php
+/**
+ * File containing the Email_List_Table_Actions class.
+ *
+ * @package sensei
+ */
+
+namespace Sensei\Internal\Emails;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * The class responsible for handling the email list table actions.
+ *
+ * @internal
+ *
+ * @since $$next-version$$
+ */
+class Email_List_Table_Actions {
+	/**
+	 * Initialize the class and add hooks.
+	 *
+	 * @internal
+	 */
+	public function init(): void {
+		add_action( 'post_action_enable-email', [ $this, 'enable_email' ] );
+		add_action( 'post_action_disable-email', [ $this, 'disable_email' ] );
+	}
+
+	/**
+	 * Enable the email by publishing the post.
+	 *
+	 * @internal
+	 *
+	 * @param int $post_id The post ID.
+	 */
+	public function enable_email( $post_id ): void {
+		check_admin_referer( 'enable-email-post_' . $post_id );
+		$this->validate_request( $post_id );
+
+		wp_publish_post( $post_id );
+
+		$this->redirect_back();
+	}
+
+	/**
+	 * Disable the email by changing the post status to "draft".
+	 *
+	 * @internal
+	 *
+	 * @param int $post_id The post ID.
+	 */
+	public function disable_email( $post_id ): void {
+		check_admin_referer( 'disable-email-post_' . $post_id );
+		$this->validate_request( $post_id );
+
+		wp_update_post(
+			[
+				'ID'          => $post_id,
+				'post_status' => 'draft',
+			]
+		);
+
+		$this->redirect_back();
+	}
+
+	/**
+	 * Ensures the request is valid and the user has permission.
+	 * If the request is not valid, the method will exit with a message.
+	 *
+	 * @param int $post_id The post ID.
+	 */
+	private function validate_request( $post_id ): void {
+		if ( ! current_user_can( 'manage_sensei' ) ) {
+			wp_die( esc_html__( 'Insufficient permissions', 'sensei-lms' ) );
```

Comment:
Just wondering if it is possible not to call `die` here?
It makes the class (and probably its users) untestable (in terms of unit tests) 🤔

Reply:
Hey Dmitry, I think it's OK, I'm following the same conventions WordPress uses in core.

About testing - you can test `wp_die`. When called from the test suite, the handler is replaced and [it throws a `WPDieException`](https://github.com/WordPress/wordpress-develop/blob/3b4132d6c096a8a17fa52b9e00a201985a243406/tests/phpunit/includes/abstract-testcase.php#L505-L520) instead of calling `die`.

Let me know if you think this can be improved.